examples/image.md

```diff
@@ -0,0 +1,234 @@
+
+Following is a snippet that would log a single image for 1 step in temp directory.
+
+```cpp
+#include <mlboard/mlboard.hpp>
+#include <iostream>
+#include <chrono> 
+#include <ctime> 
+#include <future>
+
+int main()
+{
+  std::chrono::time_point<std::chrono::system_clock> start, end; 
+  start = std::chrono::system_clock::now(); 
+  mlboard::FileWriter f1("temp");
+  
+  std::ifstream fin("./assets/single_image.jpg", std::ios::binary);
+  std::ostringstream ss;
+  ss << fin.rdbuf();
+  std::string image(ss.str());
+  fin.close();
+  mlboard::SummaryWriter<mlboard::FileWriter>::Image(
+        "Test Image", 1, image, 512, 512, 3, f1, "Sample Image",
+        "This is a Sample image logged using mlboard.");
+  f1.Close();
+  end = std::chrono::system_clock::now(); 
+  std::chrono::duration<double> elapsed_seconds = end - start; 
+  std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
+  
+  std::cout << "finished computation at " << std::ctime(&end_time) 
+            << "elapsed time: " << elapsed_seconds.count() << "s\n"; 
+}
+```
+
+The output would be similar to: 
+
+<p>
+<img src = "assets/single_image.jpg" width = "800" height = "400"/>
+</p>
+
+### 2. Multiple Images
+
+To log multiple image at an instance you can use the following API:
+
+```cpp
+void SummaryWriter<Filewriter>::Image(const std::string& tag,
+                                      int step,
+                                      const std::vector<std::string>& encodedImages,
+                                      int height,
+                                      int width,
+                                      mlboard::Filewriter& fw,
+                                      const std::string& displayName,
+                                      const std::string& description)
+```
+
+The API accepts `tag`, `step`, `encodedImage` (A vector of Image data in encoded format), the metadata about images (height and width) along with `mlboard::Filewriter` object, Name and description.
+
+Following is a snippet that would log a multiple image for 1 step in temp directory.
+
+```cpp
+#include <mlboard/mlboard.hpp>
+#include <iostream>
+#include <chrono> 
+#include <ctime> 
+#include <future>
+
+int main()
+{
+  std::chrono::time_point<std::chrono::system_clock> start, end; 
+  start = std::chrono::system_clock::now(); 
+  mlboard::FileWriter f1("temp");
+  
+  std::ifstream fin("./assets/multiple_image.jpg", std::ios::binary);
+  std::ostringstream ss;
+  std::vector<std::string>encodedImages;
+  ss << fin.rdbuf();
+  encodedImages.push_back(ss.str());
+  fin.close();
+  fin.open("./assets/single_image.jpg", std::ios::binary);
+  ss << fin.rdbuf();
+  encodedImages.push_back(ss.str());
+  ss.str("");
+  fin.close();
+  mlboard::SummaryWriter<mlboard::FileWriter>::Image(
+        "Test Image", 1, encodedImages, 512, 512, f1, "Sample Image",
```

Comment:
Hey, 512 and 512 is the size of the input image? Or it is of the image which will get displayed in the dashboard? I saw the assets folder the image dimensions are quite different.

Reply:
it is of the image which will get displayed in the dashboard, Currently you can give 0,0 also it doesn't make a difference, It will only make a difference I think so from next release because currently tensor board ignores those values, I have no clue why

Reply:
Okay, If I am not wrong. You are only retrieving the height and width of the image from the `imageInfo`object. I think it's better we remove the use of ImageInfo from the API because the user can also enter the height and width manually. Let me know what you think.

Reply:
mlpack::data::Save(fileNames, matrix, info, false);


This line uses it, we need the channels to save it and then while reading it back we need not need it

Reply:
Oh yeah, Thanks for the info. That line got overlooked.